### PR TITLE
Build package using pyproject.toml

### PR DESCRIPTION
**Pull request by RommelLayco**
Migrate the building of the package to use a pyproject.toml to meet pep-0621

### CODE REVIEW

**Review by jelmer — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,81 @@
...
+    "pytest-timeout",
+    "pytest-xdist",
+    "responses",
+    "time-machine",
+    "types-PyMySQL",
+    "types-requests",
+]
+
+[project.urls]
+"Homepage" = "https://github.com/aiven/myhoard/"
```
Comment (jfg-aiven) [inline pyproject.toml:66]: This redirects to https://github.com/Aiven-Open/myhoard/, shouldn't this directly go there ?

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,81 @@
...
+"Bug Tracker" = "https://github.com/aiven/myhoard/issues"
```
[the remark this answers is not in the document]
Comment (jelmer) [reply pyproject.toml:67]: I think the Aiven-Open org didn't exist when this PR was created, but it does indeed make sense to update the URLs.

[the remark this answers is not in the document]
Comment (RommelLayco) [reply pyproject.toml]: i have updated

File: pyproject.toml (hunk 1 above)
Comment (RommelLayco) [reply pyproject.toml]: i have updated

**Review by facetoe — APPROVED** [review]

**Pull request merged**